polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1144,13 +1156,24 @@ impl<Config: config::Config> XcmExecutor<Config> {
 			InitiateTeleport { assets, dest, xcm } => {
 				let old_holding = self.holding.clone();
 				let result = Config::TransactionalProcessor::process(|| {
+					// When not using `PayFees`, nor `JIT_WITHDRAW`, transport fees are paid from
+					// transferred assets.
+					let maybe_transport_fee_from_holding = if self.fees.is_empty() && !self.fees_mode.jit_withdraw {
+						self.get_delivery_fee_from_holding(&assets, &dest, FeeReason::InitiateTeleport, &xcm)?
+					} else {
+						None
+					};
 					let assets = self.holding.saturating_take(assets);
```

Comment:
shouldn't we also substract delivery fees here?, e.g.:
```
let assets = self.holding.saturating_take(assets - maybe_transport_fee_from_holding);
```

Reply:
hmm.. yes, good catch!

most times it doesn't matter since `assets` is most times a wildcard or some sum where we saturate when we take.

but there could be usecases where someone wants to send exactly `N` tokens while keeping other `M` for some subsequent instruction - so it makes sense to send `N-transport` and leave `M` for next instruction rather than how the code is now where we send `N` and leave `M-transport`.

I'll make the change.

Reply:
fixed here https://github.com/paritytech/polkadot-sdk/pull/4834/commits/275295f4622509927c99fd6bdfdff537cf2103c2